etcdserver/security/security.go

```diff
@@ -116,7 +116,6 @@ func NewStore(server doer, timeout time.Duration) *Store {
 		server:  server,
 		timeout: timeout,
 	}
-	s.ensureSecurityDirectories()
 	s.enabled = s.detectSecurity()
```

Comment:
Yes. And it is hard to tell when it catches up with the cluster progress. Getting it from store each time is one method to ensure its correctness.